session/pingpong/consumer_balance_tracker.go

```diff
@@ -129,8 +125,13 @@ func (cbt *ConsumerBalanceTracker) handleUnlockEvent(id string) {
 	cbt.ForceBalanceUpdate(identity)
 }
 
-func (cbt *ConsumerBalanceTracker) handleGrandTotalRecovered(ev GrandTotalRecovered) {
-	cbt.ForceBalanceUpdate(ev.Identity)
+func (cbt *ConsumerBalanceTracker) handleGrandTotalChanged(ev AppEventGrandTotalChanged) {
```

Comment:
I suggest to pass to what amount it changed in `AppEventGrandTotalChanged `, and you would need to call `consumerGrandTotalGetter.Get()` or couple to this storage at all

Reply:
I'm already passing in the amount.